api_fhir_r4/paginations.py

```diff
@@ -68,3 +72,32 @@ def exclude_query_parameter_from_url(self, url):
             from urlparse import urlparse  # Python 2
         o = urlparse(url)
         return o._replace(query=None).geturl()
+
+    def paginate_queryset(self, queryset, *args, **kwargs):
+        if hasattr(queryset, 'count'):
+            queryset = CachedCountQueryset(queryset)
+        return super().paginate_queryset(queryset, *args, **kwargs)
+ 
+def CachedCountQueryset(queryset, timeout=60*60, cache_name='default'):
```

Comment:
@dragos-dobre this is important for performance, applies only if the query is the same: will never apply if the "check validity" filter is applied because it adds the query timestamp up to the millisec